learning/inference.py

```diff
@@ -4,6 +4,8 @@
 from contextlib import contextmanager
 from pathlib import Path
 from time import sleep
+import math
+import boxenv
```

Comment:
Are these inputs being used? Check your VSCode extensions, I think `ruff` should catch this.